### PR TITLE
Added functionality to export asset details in csv format

### LINKED ISSUE #5483: Add CSV support for the asset import/export
**Describe the bug**

We don't support the CSV import/export of asset for a facility

**To Reproduce**

1. Go to https://care.coronasafe.in/assets?page=2&limit=18&search=
2. See error

**Expected behavior**

We should be able to use CSV to import and export bulk assets

**Screenshots**

image

### PR DESCRIPTION
**Pull request by patelaryan7751**
### <samp>🤖 Generated by Copilot at c82f926</samp>

Added CSV export option for assets list and renamed JSON export option in `AssetsList.tsx`. Used a new function to convert API data to CSV format and added a button with conditional logic.

## Proposed Changes

- Fixes #5483 
- Added functionality to export asset details in CSV format
image
image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at c82f926</samp>

* Add a CSV export option for the assets list ([link](https://github.com/coronasafe/care_fe/pull/5566/files?diff=unified&w=0#diff-d61b507f07891c695d44e81effbf8f1da3a8667b0379dcfbd98c5761aa124f49L318-R361))
* Define a function to convert an API response object to a CSV string ([link](https://github.com/coronasafe/care_fe/pull/5566/files?diff=unified&w=0#diff-d61b507f07891c695d44e81effbf8f1da3a8667b0379dcfbd98c5761aa124f49L110-R127))

### CODE REVIEW

File: src/Components/Assets/AssetsList.tsx [hunk 1]
```diff
@@ -315,7 +335,24 @@ const AssetsList = () => {
                     },
                   },
                   {
-                    label: "Export Assets",
+                    label: "Export Assets in CSV",
```
Comment (rithviknishad) [inline src/Components/Assets/AssetsList.tsx]:
```suggestion
                    label: "Export Assets as CSV",
```

File: src/Components/Assets/AssetsList.tsx [hunk 2]
```diff
@@ -315,7 +335,24 @@ const AssetsList = () => {
...
+                    type: "csv",
+                    parse: exportToCSV,
+                    filePrefix: `assets_${facility?.name && ""}`,
+                    options: {
+                      icon: <CareIcon className="care-l-export" />,
+                      disabled: totalCount === 0 || !authorizedForImportExport,
+                    },
+                  },
+                  {
+                    label: "Export Assets in JSON",
```
Comment (rithviknishad) [inline src/Components/Assets/AssetsList.tsx]:
```suggestion
                    label: "Export Assets as JSON",
```

**Review by rithviknishad — APPROVED** [review]

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@patelaryan7751 can you do the CSV export in the backend instead?

All other CSV exports and formatting is handled in the backend. 

cc: @nihal467

**Comment by patelaryan7751** [comment]
yeah sure @rithviknishad i am creating a backend issue for this

**Comment by patelaryan7751** [comment]
I have created a backend issue for the same https://github.com/coronasafe/care/issues/1327

**Comment by nihal467** [comment]
it is already solved in the platform ( #6262 ), so closing the related PR & issue

**Pull request closed without merging**